### PR TITLE
Port the `ImageClassifier` guide to `keras_core`

### PR DESCRIPTION
**Pull request by tirthasheshpatel**
This PR ports the [ImageClassifier guide on keras.io](https://keras.io/guides/keras_cv/classification_with_keras_cv/) to `keras_core`. It uses the `keras_cv.backend` module to import Keras which provides support for both `tf.keras` and `keras_core` in a backend-agnostic manner.

To run the guide on Colab, use the [`core` branch from Ian's fork](https://github.com/ianstenbit/keras-cv/tree/core) of Keras-CV where we are porting backbones to `keras_core`.

cc @ianstenbit

### CODE REVIEW

File: examples/keras_io/vision/image_classifier.py [hunk 1]
```diff
@@ -0,0 +1,679 @@
+# -*- coding: utf-8 -*-
+"""Classification in Keras-CV
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1qvQbaRuuMKJzp7cYgEQ7lY6kvBzglIqj
+
+# Classification with KerasCV
+
```
Comment (fchollet) [inline examples/keras_io/vision/image_classifier.py:10]: Remove the first few lines above

File: examples/keras_io/vision/image_classifier.py [hunk 2]
```diff
@@ -0,0 +1,679 @@
...
+**Author:** [lukewood](https://lukewood.xyz)<br>
```
Comment (fchollet) [inline examples/keras_io/vision/image_classifier.py:11]: This is using markdown formatting -- don't. See other files for examples

File: examples/keras_io/vision/image_classifier.py [hunk 3]
```diff
@@ -0,0 +1,679 @@
...
+- Training a image classifier from scratch
+
+We use Professor Keras, the official Keras mascot, as a
+visual reference for the complexity of the material:
+
+![](https://storage.googleapis.com/keras-nlp/getting_started_guide/prof_keras_evolution.png)
+
+## Multi-Backend Support
+
+Keras-CV's `ImageClassifier` model supports several backends like Jax, PyTorch, and TensorFlow with the help of `keras_core`. To enable multi-backend support in Keras-CV, set the `KERAS_CV_MULTI_BACKEND` environment variable. We can then switch between different backends by setting the `KERAS_BACKEND` environment variable. Currently, `tensorflow`, `jax`, and `torch` are supported.
```
Comment (fchollet) [inline examples/keras_io/vision/image_classifier.py:36]: - Keep lines under 80 chars
- Refer to KerasCV as "KerasCV"

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

**Comment by tirthasheshpatel** [comment]
@fchollet I don't have the permission to push to the upstream branch directly anymore, so, I have created a #608 on this branch addressing your comments here.

File: examples/keras_io/vision/image_classifier.py (hunk 1 above)
Comment (tirthasheshpatel) [reply examples/keras_io/vision/image_classifier.py:10]: Done.

File: examples/keras_io/vision/image_classifier.py (hunk 2 above)
Comment (tirthasheshpatel) [reply examples/keras_io/vision/image_classifier.py:11]: Done in #608.

File: examples/keras_io/vision/image_classifier.py (hunk 3 above)
Comment (tirthasheshpatel) [reply examples/keras_io/vision/image_classifier.py:36]: Done in #608 with the exception of a few links that can't be wrapped to fit in 80 chars character limit.

**Comment by fchollet** [comment]
Let's carry on in the other PR then!

**Pull request closed without merging**